core/src/main/java/org/elasticsearch/script/ScriptEngineRegistry.java

```diff
@@ -67,26 +72,42 @@ public ScriptEngineRegistry(Iterable<ScriptEngineRegistration> registrations) {
         return registeredLanguages;
     }
 
+    Map<String, ScriptMode> getDefaultInlineScriptModes() {
+        return this.defaultInlineScriptModes;
+    }
+
     public static class ScriptEngineRegistration {
         private final Class<? extends ScriptEngineService> scriptEngineService;
-        private final List<String> scriptEngineLanguages;
+        private final String scriptEngineLanguage;
+        private final ScriptMode defaultInlineScriptMode;
 
-        public ScriptEngineRegistration(Class<? extends ScriptEngineService> scriptEngineService, List<String> scriptEngineLanguages) {
+        public ScriptEngineRegistration(Class<? extends ScriptEngineService> scriptEngineService, String scriptEngineLanguage) {
+            // Default to "off/disabled" if not specified
```

Comment:
This feels like it should be in the javadoc for this constructor or the constructor shouldn't exist at all?